string data

---------------
string result
bool success
